sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -150,6 +150,7 @@ case class LogicalRDD(
   }
 
   override lazy val constraints: ExpressionSet = originConstraints.getOrElse(ExpressionSet())
+    .filterNot(SubqueryExpression.hasSubquery)
```

Comment:
can we add some comments?

Reply:
It makes sense as the checkpointed query has already been executed and we can't reuse the subquery result in the current query execution.

Reply:
Done!